### PR TITLE
add protected flash writes/erases for when SD is in use

### PR DESCRIPTION
**Pull request by jamesadevine**
Soft device requires control over everything...

### CODE REVIEW

File: libs/settings/NRF52Flash.cpp [hunk 1]
```diff
@@ -56,21 +98,40 @@ int ZFlash::writeBytes(uintptr_t dst, const void *src, uint32_t len) {
...
-            *dp++ = v;
-            waitForLast();
-        } else {
-            dp++;
+        while(1)
+        {
+            if (sd_flash_write((uint32_t *)dp, (uint32_t *)sp, len) == NRF_SUCCESS)
+                break;
+
+            system_timer_wait_ms(10);
```
Comment (mmoskal) [inline libs/settings/NRF52Flash.cpp:111]: I think this may be too slow for anything useful. Maybe you can spin on flash_op_complete?

Comment (jamesadevine) [reply libs/settings/NRF52Flash.cpp:111]: If the previous call does not go through then flash_op_complete will never trigger as the async callback will never take place. This is the same code that is used in microbit codal now!

Comment (mmoskal) [reply libs/settings/NRF52Flash.cpp:111]: so you're saying the wait_ms is usually not hit?

Comment (jamesadevine) [reply libs/settings/NRF52Flash.cpp:111]: From my observations during my primitive testing, I have not yet hit this condition. I have only ever gotten NRF_SUCCESS back from these calls.

**Review by mmoskal — APPROVED** [review]

**Comment by jamesadevine** [comment]
@mmoskal Can we merge?

**Comment by mmoskal** [comment]
I mean it was approved...

**Pull request merged**